dvc/objects.py

```diff
@@ -19,14 +19,17 @@ class ObjectFormatError(ObjectError):
     pass
 
 
-def _get_hash(path_info, tree, name, **kwargs):
+def _get_hash(odb, path_info, tree, name, **kwargs):
     if tree.PARAM_CHECKSUM == name:
         return tree.get_hash(path_info, **kwargs)
 
     if name == "md5":
         from dvc.hash_info import HashInfo
+        from dvc.odb.versions import ODB_VERSION
         from dvc.utils import file_md5
 
+        if odb.version == ODB_VERSION.V1:
+            kwargs["enable_d2u"] = True
         return HashInfo("md5", file_md5(path_info, tree, **kwargs))
```

Comment:
stuff under `if name == "md5"` is defunct, just added to clarify the intentions.

We should actually do `tree.get_hash(path_info, name, **kwargs)` and so then, for example, LocalTree.get_hash will see that we ask it about `md5` and will use `file_md5` and if it sees `dos2unixmd5` - `file_dos2unixmd5`